Formula/git-annex.rb

```diff
@@ -14,17 +14,42 @@ class GitAnnex < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux: "8e82e574c6170247e9805813625e28ce24d0eae368d6151f2ce05b56f497b1f2"
   end
 
-  depends_on "cabal-install" => :build
-  depends_on "ghc" => :build
+  if Hardware::CPU.arm?
+    depends_on "ghcup" => :build
+    depends_on "llvm@12" => :build
+  else
+    depends_on "cabal-install" => :build
+    depends_on "ghc" => :build
+  end
```

Comment:
I'm not sure where this is going. Is this not an indication that our GHC is broken?

Reply:
Only testing different possibilities right now and trying to eliminate some variables. I cannot reproduce failures locally, so slowly experimenting on CI runners.

I plan to remove if I can figure out what is the actual issue since similar failures are still happening with pre-built.

Currently considering Homebrew's environment. Some previous errors were running `llc`/`opt`. Current error for me is running `clang`, but not sure if it is actually issue with finding `clang` in `PATH` or the args passed to `clang`.

Reply:
Ok, I'm happy to help look into this more tomorrow if you haven't figured it out by then.

Reply:
Thanks. I'm out of ideas and will need to pass this off.

I tried disabling some features and doing a minimal build based on upstream instructions, but it still failed. Also tried to serialize build to see if I could narrow issue, but Monterey and Big Sur are failing at different locations:
- Monterey fails on `[554 of 567] Compiling Annex.Drop` while running `llc`:
  ```
  llc -O2 -enable-tbaa '-relocation-model=pic' '-mcpu=apple-a7' '-mattr=+fp-armv8,+neon,+crypto,+zcm,+zcz,+sha2,+aes' /private/tmp/ghc46922_0/ghc_6332.bc -o /private/tmp/ghc46922_0/ghc_6333.lm_s
  ```
- Big Sur fails on `[545 of 567] Compiling Command.Adjust` while running `llc`:
  ```
  llc -O2 -enable-tbaa '-relocation-model=pic' '-mcpu=apple-a7' '-mattr=+fp-armv8,+neon,+crypto,+zcm,+zcz,+sha2,+aes' /private/tmp/ghc41959_0/ghc_6228.bc -o /private/tmp/ghc41959_0/ghc_6229.lm_s
  ```

On other runs, it sometimes failed running `opt`. Additional logging (`-v4` or `-v5`) generate massive GB-sized logs files and slow run over timeout limit, so haven't had chance to get a run with those.

Reply:
I think I've figured it out: #99021